### PR TITLE
[9.0] [NEW] stock_available_unreserved

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
# 
# Stock Available Unreserved

This module allows users to check the quantity of a stocked product that is
available on-hand, and that has not yet been reserved for use anywhere else.

This key figure is very important during the monitoring of the warehouse
execution, because it assists users to ensure that the flow of products will
not be stuck due to a sudden unavailability of stock.

If the warehouse personnel ensures that the unreserved quantity on hand > 0,
then nobody will be stuck in pickings or manufacturing orders waiting for
the availability of unreserved stock.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
cc @sbidoul

File: stock_available_unreserved/models/product.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        domain_quant += domain_products
+
+        domain_quant.append(('reservation_id', '=', False))
+
+        domain_quant += domain_quant_loc
+
+        return domain_quant
+
+    @api.multi
+    def _product_available_not_res_hook(self, quants):
```
Comment (AaronHForgeFlow) [inline stock_available_unreserved/models/product.py:107]: I don't understand why this is necessary. The hook is only at the end of the '_product_available' method so we just need to call super to inherit, don't we?

**Review by AaronHForgeFlow — COMMENTED** [review]
Runbot is not available but tested on local :+1:

**Review by Cedric-Pigeon — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
Thanks for your feedback @Cedric-Pigeon

**Comment by pedrobaeza** [comment]
Can you provide tests?

File: stock_available_unreserved/models/product.py (hunk 1 above)
Comment (JordiBForgeFlow) [reply stock_available_unreserved/models/product.py:107]: Not really because you do not have quants when you inherit

**Comment by rousseldenis** [comment]
Even if this one was pushed some months ago, maybe more reviews should be preferable. Moreover, one function is not covered by tests.

**Comment by JordiBForgeFlow** [comment]
Thanks @rousseldenis. We will make a separate PR to cover with tests the missing function.
cc @mreficent

**Comment by rousseldenis** [comment]
@jbeficent Otherwise :+1:  ;-)

**Comment by MiquelRForgeFlow** [comment]
@rousseldenis 
The method <code>action_open_quants_unreserved()</code> is an action that returns a simple dictionary from <code>product.template</code>. Thus, I think that the method is not necessary to be covered in the test but just tested manually.

**Pull request merged**